### PR TITLE
construct a user key map

### PR DESCRIPTION
**Pull request by george-angel**

### CODE REVIEW

File: sync.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	Keys []string `json:"keys"`
+}
+
+func authmap(adminClient *http.Client, groups []string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var rtnGroups []Group
+		for _, g := range groups {
+			var memList GoogleMemberList
+			group := Group{Name: g, Keys: []string{}}
+			req, _ := http.NewRequest(http.MethodGet, fmt.Sprintf(adminGroupMembersURI, g), nil)
```
Comment (catalin-ilea) [inline sync.go]: Can you add some error handling?

File: sync.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+			resp, _ := adminClient.Do(req)
```
Comment (catalin-ilea) [inline sync.go]: Close resp body so that tcp connections are reused

File: sync.go (hunk 1 above)
Comment (george-angel) [reply sync.go]: Yes, later.

File: sync.go (hunk 2 above)
Comment (george-angel) [reply sync.go]: Do I still need to read it all like so? 

https://github.com/utilitywarehouse/ssh-key-manager/blob/master/main.go#L73-L76

Comment (catalin-ilea) [reply sync.go]: That's how I usually do it. `io.Copy(ioutil.Discard, resp.Body)` might be redundant though.

File: sync.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+			buf := new(bytes.Buffer)
```
Comment (catalin-ilea) [inline sync.go]: You should consider using `json.NewDecoder(resp.Body)` instead. That does its own buffering.

**Comment by catalin-ilea** [comment]
Any tests? :)

**Comment by catalin-ilea** [comment]
```
main.go:58:2:warning: struct field IdToken should be IDToken (golint)
main.go:189:3:warning: var userKeysUri should be userKeysURI (golint)
```

**Comment by george-angel** [comment]
No, sorry, I needed to preface this properly with, this is work half way through, just obviously wrong things please, I'll validation/error handling later as a lot of this will change.

**Pull request merged**